cocos/render-scene/core/pass.ts

```diff
@@ -41,6 +41,7 @@ import { MacroRecord, MaterialProperty, customizeType, getBindingFromHandle, get
 import { RenderPassStage, RenderPriority } from '../../rendering/define';
 import { InstancedBuffer } from '../../rendering/instanced-buffer';
 import { BatchedBuffer } from '../../rendering/batched-buffer';
+import { ProgramLibrary } from '../../rendering/custom/private';
```

Comment:
This is the private interface of custom-pipeline，it contains very little information.
I think after compilation, it will contains no code.
I will check compiled code to ensure that.

Reply:
I have checked the compiled cc.js, and `ProgramLibrary` has not appeared. I believe it is removed by `tsc`.